CharlieBackend.Data/Repositories/Impl/AccountRepository.cs

```diff
@@ -101,12 +101,12 @@ public async Task<bool> DisableAccountAsync(long id)
         {
             var foundAccount = await _applicationContext.Accounts
                     .FirstOrDefaultAsync(account => account.Id == id);
-            if (foundAccount == null)
+            if (foundAccount == null || (bool)!foundAccount.IsActive)
```

Comment:
please make this property non-nullable, since in our schema we have it so, and remove cast to bool. (By the way, you would get InvalidCastException if IsActive contained null)